vendor/bat-native-ledger/src/bat/ledger/internal/wallet_provider/transfer.cc

```diff
@@ -0,0 +1,115 @@
+      // The least amount one can tip is 0.25, where:
+      //   - 0.2375 (95%) is the actual contribution, and
+      //   - 0.0125 (== 1/80) (5%) is the fee
+      // Rounding to nearest ten-thousandth:
+      base::NumberToString(std::round(amount * 10000) / 10000),
+      base::BindOnce(&Transfer::CommitTransaction, base::Unretained(this),
+                     std::move(callback)));
+}
+
+void Transfer::MaybeCreateTransaction(
+    const std::string& contribution_id,
+    const std::string& destination,
+    const std::string& amount,
+    MaybeCreateTransactionCallback callback) const {
+  ledger_->database()->GetExternalTransaction(
+      contribution_id, destination,
+      base::BindOnce(&Transfer::OnGetExternalTransaction,
+                     base::Unretained(this), std::move(callback),
+                     contribution_id, destination, amount));
+}
+
+void Transfer::OnGetExternalTransaction(
+    MaybeCreateTransactionCallback callback,
+    std::string&& contribution_id,
+    std::string&& destination,
+    std::string&& amount,
+    absl::optional<mojom::ExternalTransactionPtr> transaction) const {
+  if (!transaction) {
+    return std::move(callback).Run(nullptr);
+  }
+
+  if (*transaction) {
+    return std::move(callback).Run(std::move(*transaction));
+  }
+
+  *transaction = mojom::ExternalTransaction::New(
+      "" /* transaction_id - to be generated */, std::move(contribution_id),
+      std::move(destination), std::move(amount));
+
+  CreateTransaction(base::BindOnce(&Transfer::SaveExternalTransaction,
+                                   base::Unretained(this), std::move(callback)),
+                    std::move(*transaction));
+}
+
+void Transfer::SaveExternalTransaction(
+    MaybeCreateTransactionCallback callback,
+    mojom::ExternalTransactionPtr transaction) const {
+  if (!transaction) {
+    return std::move(callback).Run(nullptr);
+  }
+
+  DCHECK(!transaction->transaction_id.empty());
+
+  auto on_save_external_transaction = base::BindOnce(
+      &Transfer::OnSaveExternalTransaction, base::Unretained(this),
+      std::move(callback), transaction->Clone());
+
+  ledger_->database()->SaveExternalTransaction(
+      std::move(transaction), std::move(on_save_external_transaction));
+}
+
+void Transfer::OnSaveExternalTransaction(
+    MaybeCreateTransactionCallback callback,
+    mojom::ExternalTransactionPtr transaction,
+    mojom::Result result) const {
+  if (result != mojom::Result::LEDGER_OK) {
+    BLOG(0, "Failed to save external transaction!");
+    return std::move(callback).Run(nullptr);
+  }
+
+  std::move(callback).Run(std::move(transaction));
+}
+
+void Transfer::CreateTransaction(
+    MaybeCreateTransactionCallback callback,
+    mojom::ExternalTransactionPtr transaction) const {
+  DCHECK(transaction);
+  DCHECK(transaction->transaction_id.empty());
+
+  transaction->transaction_id = base::GenerateGUID();
+
+  base::SequencedTaskRunnerHandle::Get()->PostTask(
```

Comment:
There was a recent change to this API. See here: https://github.com/brave/brave-core/pull/16794

Reply:
Thanks, changed to:
```
base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
    FROM_HERE, base::BindOnce(std::move(callback), std::move(transaction)));
```